src/cluster_slot_stats.c

```diff
@@ -148,10 +148,12 @@ static void clusterSlotStatsUpdateNetworkBytesOutForReplication(long long len) {
     client *c = server.current_client;
     if (c == NULL || !canAddNetworkBytesOut(c)) return;
 
+    // we multiply the bytes len by the number of replicas to account for us broadcasting to multiple replicas at once
+    len *= (long long)listLength(server.replicas);
     serverAssert(c->slot >= 0 && c->slot < CLUSTER_SLOTS);
     serverAssert(nodeIsPrimary(server.cluster->myself));
-    if (len < 0) serverAssert(server.cluster->slot_stats[c->slot].network_bytes_out >= (uint64_t)llabs(len));
-    server.cluster->slot_stats[c->slot].network_bytes_out += (len * listLength(server.replicas));
+    serverAssert(len >= 0 || server.cluster->slot_stats[c->slot].network_bytes_out >= (uint64_t)-len);
```

Comment:
Can you add a comment here? Or maybe even better: a comment before the function, to explain what it means to call the function with a negative len. Normally you can't send a negative number of bytes over a network, so I think a comment would be useful to explain what it means.

```C
    /* If len is negative, check bla bla bla ... */
    serverAssert(len >= 0 || server.cluster->slot_stats[c->slot].network_bytes_out >= (uint64_t)-len);
```

Reply:
There we go, added a more verbose comment for that as well now.